iot-hub/Samples/service/registry/registry_bulk_sample.js

```diff
@@ -1,140 +1,155 @@
-  process.exit(-1);
-}
-
-var registry = iothub.Registry.fromConnectionString(iothubConnectionString);
-var blobSvc = azureStorage.createBlobService(storageConnectionString);
-
-var startDate = new Date();
-var expiryDate = new Date(startDate);
-expiryDate.setMinutes(startDate.getMinutes() + 100);
-startDate.setMinutes(startDate.getMinutes() - 100);
-
-var inputSharedAccessPolicy = {
-  AccessPolicy: {
-    Permissions: 'rl',
-    Start: startDate,
-    Expiry: expiryDate
-  },
-};
-
-var outputSharedAccessPolicy = {
-  AccessPolicy: {
-    Permissions: 'rwd',
-    Start: startDate,
-    Expiry: expiryDate
-  },
-};
-
-var inputContainerName = 'importcontainer';
-var outputContainerName = 'exportcontainer';
-var deviceFile = 'devices.txt';
-
-blobSvc.createContainerIfNotExists(inputContainerName, function (error) {
-    if(error) {
-        console.error('Could not create input container: ' + error.message);
+"use strict";
+
+var iothub = require("azure-iothub");
+var azureStorage = require("@azure/storage-blob");
+
+async function main() {
+  var iothubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+  if (!iothubConnectionString) {
+    console.log("Please set the IOTHUB_CONNECTION_STRING environment variable.");
+    process.exit(-1);
+  }
+
+  var storageConnectionString = process.env.STORAGE_CONNECTION_STRING;
+  if (!storageConnectionString) {
+    console.log("Please set the STORAGE_CONNECTION_STRING environment variable.");
+    process.exit(-1);
+  }
+
+  var registry = iothub.Registry.fromConnectionString(iothubConnectionString);
+  var blobSvc = azureStorage.BlobServiceClient.fromConnectionString(storageConnectionString);
+
+  var accountName = "<storage-account-name>";
+  var key = "<storage-account-key>";
+
+  var credential = new azureStorage.StorageSharedKeyCredential(accountName, key);
+
+  var startDate = new Date();
+  var expiryDate = new Date(startDate);
+  expiryDate.setMinutes(startDate.getMinutes() + 100);
+  startDate.setMinutes(startDate.getMinutes() - 100);
+
+  var inputContainerName = "importcontainer";
+  var outputContainerName = "exportcontainer";
+  var deviceFile = "devices.txt";
+
+  var inputSASSignatureValues = {
+    containerName: inputContainerName,
+    permissions: azureStorage.AccountSASPermissions.parse("rl"),
+    startsOn: startDate,
+    expiresOn: expiryDate,
+  };
+
+  var outputSASSignatureValues = {
+    containerName: outputContainerName,
+    permissions: azureStorage.AccountSASPermissions.parse("rwd"),
+    startsOn: startDate,
+    expiresOn: expiryDate,
+  };
+
+  var inputContainerClient = blobSvc.getContainerClient(inputContainerName);
+
+  try {
+    await inputContainerClient.createIfNotExists();
+  } catch (err) {
+    console.error("Could not create input container: " + err.message);
+  }
+
+  var inputSasToken = azureStorage.generateBlobSASQueryParameters(inputSASSignatureValues, credential).toString();
+  var inputSasUrl = inputContainerClient.url + "?" + inputSasToken;
```

Comment:
You can use ContainerClient.generateSasUrl to get the URL with SAS token, this way only STORAGE_CONNECTION_STRING is required, "accountName" or "key" won't be required.

Reply:
I tried to use it you said. but I got an error message : `Can only generate the SAS when the client is initialized with a shared key credential`

Reply:
Does STORAGE_CONNECTION_STRING use account and key? Or could you share the STORAGE_CONNECTION_STRING value (without actual credentials, only the format is enough)?